src/univariate/discrete/negativebinomial.jl

```diff
@@ -128,7 +128,7 @@ end
 
 function mgf(d::NegativeBinomial, t::Real)
     r, p = params(d)
-    return ((1 - p) * exp(t))^r / (1 - p * exp(t))^r
+    return mgf(Geometric(p), t)^r
```

Comment:
I think that's wrong (see e.g. https://en.wikipedia.org/wiki/Moment-generating_function#Examples)? My guess would be it should be something like
```suggestion
    return (p / (1 - (1 - p) * exp(t)))^r
```

Reply:
There is a subtle point here and with your other comment. Namely there are two widespread ways to parametrize the Geometric distribution, see https://en.wikipedia.org/wiki/Geometric_distribution

What is called `k-1` in the wikipedia mgf table is called `k` in `Distributions.jl`
With the Distributions.jl parametrizations `NegativeBinomial` is a convolution power in the most straight forward fashion
https://github.com/JuliaStats/Distributions.jl/blob/aca39200d7e2288b571895a09d8e0ec34b8dcf83/src/convolution.jl#L37

This translates into the mgf of NegativeBinomial being a power of that of Geometric.

Reply:
Thanks for the explanation. Also with the formula in https://github.com/JuliaStats/Distributions.jl/pull/1604/files/1b75633699749011a03a6b0ec85c335a8281073a#r945344615 both the code in the PR (calling `mgf(Geometric(p), t)`) and the suggestion above would be identical it seems.